app/src/main/java/com/quran/labs/androidquran/AdvancedQuranPreferenceActivity.java

```diff
@@ -0,0 +1,104 @@
+package com.quran.labs.androidquran;
+
+import android.Manifest;
+import android.app.Fragment;
+import android.app.FragmentManager;
+import android.content.pm.PackageManager;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v4.app.ActivityCompat;
+import android.support.v7.app.ActionBar;
+import android.support.v7.widget.Toolbar;
+import android.view.MenuItem;
+import android.widget.Toast;
+
+import com.quran.labs.androidquran.service.util.PermissionUtil;
+import com.quran.labs.androidquran.ui.QuranActionBarActivity;
+import com.quran.labs.androidquran.ui.fragment.QuranAdvancedSettingsFragment;
+import com.quran.labs.androidquran.util.AudioManagerUtils;
+import com.quran.labs.androidquran.util.QuranSettings;
+
+public class AdvancedQuranPreferenceActivity extends QuranActionBarActivity {
+
+  private static final String SI_LOCATION_TO_WRITE = "SI_LOCATION_TO_WRITE";
+  private static final int REQUEST_WRITE_TO_SDCARD_PERMISSION = 1;
+
+  private String mLocationToWrite;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    ((QuranApplication) getApplication()).refreshLocale(this, false);
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.preferences);
+
+    final Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+    toolbar.setTitle(R.string.menu_settings);
```

Comment:
Thiught about using the exsting string as we will need a localization String for any new String will do this.